storage/client/bucket.cc

```diff
@@ -0,0 +1,64 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "storage/client/bucket.h"
+#include "google/cloud/internal/throw_delegate.h"
+#include <sstream>
+#include <thread>
+
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+static_assert(std::is_copy_constructible<storage::Bucket>::value,
+              "storage::Bucket must be constructible");
+static_assert(std::is_copy_assignable<storage::Bucket>::value,
+              "storage::Bucket must be assignable");
+
+BucketMetadata Bucket::GetMetadata() {
+  // TODO(#555) - use policies to implement retry loop.
+  Status last_status;
+  for (int i = 0; i != 3; ++i) {
+    auto result = client_->GetBucketMetadata(bucket_name_);
```

Comment:
If I'm reading correctly, as of the this commit, the GetBucketMetadata method doesn't exist in the Client class yet -- did you mean to include any changes to client.h and client.cc along with this PR? Or are you saving that for a future commit?

Reply:
The code is already merged:

https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/0ab2a95d19699c93017cf53ca0068883af8e4ee9/storage/client/internal/default_client.h#L31

Reply:
Oh, it's defined in default_client.h, not client.h.  Derp.  Thanks :)